### DISCUSSION TITLE
returning files from defineEventHandler

### DISCUSSION THREAD

**Discussion by GitMcClintock** [nuxt/nuxt#19323] [Questions] [answered]
How do you return a file from the Nuxt3 server api?  For example node express has the **res.sendFile()** method.  I have searched for comprehensive documentation for Nuxt defineEventHandler() which I hoped might set out the options for returning values, but without success.

I wish to use the api to return images, which will come from a large image repository which is not visible to clients.  The api call will specify the image required, and the server api will fetch and return it, possibly resizing it in the process and/or returning it from a cache.

Any help would be appreciated!

**Comment by Baroshem** [comment, marked as answer]
Hey,

I think that you could try to use the `sendStream` method from `unjs/h3` -> https://www.jsdocs.io/package/h3#sendStream

I have not found any official examples but I am linking to one of the users examples that may help you https://github.com/lvzhenbo/nuxt-app/blob/master/server/api/epub.ts :)

**Reply by GitMcClintock** [reply]
That's perfect thank you - the Ivzhenbo example contains the line I am looking for:

`return sendStream(event, fs.createReadStream(filePath));`

Thank you!

**Reply by GitMcClintock** [reply]
Further to the above: this is actually mentioned in the documentation, at https://nuxt.com/docs/guide/directory-structure/server under "Sending Streams (Experimental)".
